### DISCUSSION TITLE
BrowseRouter is now an indispensable tool for me

### DISCUSSION THREAD

**Discussion by helios210** [nref/BrowseRouter#27] [General]
With Google increasingly cornering the browser market I find some websites just don't play nice with Firefox, but I very much want to use Firefox wherever possible, therefore being able to route only specific sites through Chrome or Edge with everything else being taken care of by the fox has been absolutely amazing for me. I would love to see an ARM native version so I can easily use it on my Surface Pro X as well, I'm not well versed in buidling applications from source, and have never done so for ARM, but increasingly tempted to do so as the emulation doesn't work for BrowseRouter and I seem unable to register it as a default browser in it's current form.

Anyway, very nice work and where can I contribute to say thanks?

**Comment by nref** [comment]
Thanks for the kind words! You can find me on BuyMeACoffee if you feel so moved. https://buymeacoffee.com/slater1

It should be easy to add an ARM version. I will look into that this week.

**Comment by helios210** [comment]
Done! Enjoy the coffee :D
________________________________

**Comment by nref** [comment]
Thank you kindly!

I've just released version 0.7.0.0 which includes an arm64 version. I would appreciate it if you could try it out - I don't have an arm device to test it on.

See `BrowseRouter.arm64.exe` at https://github.com/nref/BrowseRouter/releases/tag/0.7.0.0

**Comment by helios210** [comment]
Wow. That was insanely fast, I will happily test and report back, thanks!

**Comment by helios210** [comment]
Not off to the best start unfortunately, I'm running Win 11 on the Surface Pro X (the first one) and running into the following:

1) The --register command appears to execute, there's no error, but BrowseRouter doesn't show up in the app selected for HTTP or HTTPS.
2) Trying to run BrowseRouter.arm64.exe www.google.com from CMD does nothing at all :(

**Comment by nref** [comment]
Hey @helios210, thanks for trying it out. Just curious if you did `--register` from an elevated prompt?

**Reply by helios210** [reply]
I did indeed, also tried the test run from an elevated prompt but neither worked unfortunately

**Comment by nref** [comment]
IIRC I think Windows 11 may have removed the ability to programmatically set the browser. After running `--register`, does it appear in the list of browsers? You can search for "default apps" in the start menu.

Here's what it look like on 10.

![image](https://github.com/user-attachments/assets/0212e3f2-65c0-43c9-9d58-f9c7c187055d)

**Reply by helios210** [reply]
So in true Microsoft fashion they've taken this perfectly functional screen in Windows 10 and completely broken and over complicated it in Windows 11 - but I can get something similar by going Default By File > searching HTTP or HTTPS and seeing Firefox, selecting this and then seeing only Chrome and Edge offered in the list or "Look for an app in the Microsoft Store"

**Comment by nref** [comment]
Another thing to try, is to download `config.ini` from the releases page if you have not already and place it next to the exe. This will enable logging. Then you can potentially see what's going on in e.g `C:\Users\<user>\AppData\Local\BrowseRouter\2024-08-29.log`

**Reply by helios210** [reply]
Ah noted, I copied a previous config from one of my x64 instances of BrowseRouter so logging wasn't enabled, I'll download this new config and report back

**Comment by helios210** [comment]
Okay, update, feel like a bit of a lemming as after a reboot I did in fact get the prompt for a new default browser, however after accepting that prompt when I click a link or run www.google.com from the start menu, nothing happens. Interestingly when I click a "safelink" in Outlook I get an error that the "application wasn't found" but despite adding the log entry to my config file and rebooting, it doesn't appear to be logging correctly.

**Reply by helios210** [reply]
*correctly = at all, %localappdata%/BrowseRouter doesn't exist

**Comment by nref** [comment]
Thanks for the detailed info!

I got my hands on an arm machine and fixed at least one issue I could see. Could you try this release?

https://github.com/nref/BrowseRouter/releases/tag/0.7.1.0

**Comment by nref** [comment]
Note I can confirm that on Windows 11 after calling `--register`, you then have to go to Default apps and choose BrowserRouter as the default. I will investigate how to improve that experience. Firefox does make it pretty easy.

**Reply by helios210** [reply]
Sorry it took me a while to test the new version but I got to it and initially no dice.

However, I sync utilities across my machines using OneDrive - this includes the BrowseRouter config. This is fine for most, but on ARM some of the pathing is different, therefore I created a new config for the Surface with the correct pathing and voila, it just works!

This was made much faster to diagnose by the logging functionality which pointed me to the issue immediately, so thanks for that, though one suggestion might be to prompt with an error if the specified browser cannot be found.

Point of the story though is that it works, and I'm one very happy camper, many thanks!
